app/models/subjects/ucas_subject_to_dfe_subject_mapping.rb

```diff
@@ -0,0 +1,40 @@
+module Subjects
+  class UCASSubjectToDFESubjectMapping
+    def initialize(ucas_input_subjects_or_options_hash, resulting_dfe_subject)
```

Comment:
I still find this a bit confusing. Given the name of this, I'd expect to be able to pass in a UCAS subject name and have a DfE subject name returned. Could we do that and change how we call this?

Reply:
Sure – I will see how I can rework this to make it clearer.

Reply:
Having just spoken in IRL, I want to build on top of this PR so I'll merge this and address the feedback in a separate PR, since this seems like an improvement over what we had previously.